packages/hoppscotch-app/pages/profile.vue

```diff
@@ -273,6 +355,39 @@ const sendEmailVerification = () => {
     })
 }
 
+const adapter = new ShortCodeListAdapter(true)
+const adapterLoading = useReadonlyStream(adapter.loading$, false)
+const adapterError = useReadonlyStream(adapter.error$, null)
+const myShortCodes = useReadonlyStream(adapter.shortCodes$, [])
+
+onLoggedIn(() => {
+  adapter.initialize()
+})
+
+const loading = computed(
+  () => adapterLoading.value && myShortCodes.value.length === 0
+)
+
+const deleteShortCode = (codeID: string) => {
+  pipe(
+    backendDeleteShortcode(codeID),
+    TE.match(
+      (err) => {
+        toast.error(`${t("error.something_went_wrong")}`)
+        console.error(err)
+      },
+      () => {
+        toast.success(`${t("shortcodes.deleted")}`)
+        refetchShortcodes()
```

Comment:
Do we have to refetch the entire list again just for deletion ? The adapter should handle deletion without refetching everything.